prefect_dask/task_runners.py

```diff
@@ -276,18 +276,24 @@ async def submit(
         else:
             dask_key = key
 
-        self._dask_futures[key] = self._client.submit(
-            call.func,
-            key=dask_key,
-            # Dask defaults to treating functions are pure, but we set this here for
-            # explicit expectations. If this task run is submitted to Dask twice, the
-            # result of the first run should be returned. Subsequent runs would return
-            # `Abort` exceptions if they were submitted again.
-            pure=True,
-            **call_kwargs,
+        self._dask_futures[key] = distributed.as_completed(
```

Comment:
We should probably do this conversion in `wait` rather than here.

Reply:
Ah nevermind I see your comment at https://github.com/PrefectHQ/prefect-dask/pull/102#discussion_r1212649749